### PR TITLE
Wishlist

### PR DESCRIPTION
**Pull request by BALAJI141031**
wishlist feat added:
files to check:
1.dry providers
2.wishlist route==>adding to wishlist and move to cart remove from wishlist

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
   return (
-    <div className="App">
-      
-    </div>
+    <>  
+      <Routes>
+      <Route path="/" element={<LandingRoute/>}/>
+      <Route path="/products" element={<Products/>}/>
+      <Route path='/cart' element={<Cart/>}/>
+      <Route path='/wishlist' element={<Wishlist/>}/>
```
Comment (tarunsankhla) [inline src/App.js]: The string of pat you can create const in a file an maintain there

File: src/App.js [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+      <Route path="/mock" element={<Mockman />}/>
```
Comment (tarunsankhla) [inline src/App.js]: Use * path for not found or for now just redirect them to home page

File: src/components/cards/productCard.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        switch(type){
+          case "navigate_to_cart":
+            return gotoCart(navigate)
+          default:
+            return await addItemToCart(payload,dispatchCart,setButtonText,cartCount)
+        }
+        
+      }catch(e){
+        throw e
+      }
```
Comment (tarunsankhla) [inline src/components/cards/productCard.js]: In catch throw error you should handle this

File: src/components/cards/textCard.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import './index.css'
+
+const TextCard=(props)=>{
+    const {textObj}=props
+    const {aboutUs}=textObj
```
Comment (tarunsankhla) [inline src/components/cards/textCard.js]: You could directly do { textObj:{aboutus}}=props

File: src/components/cards/wishlistCard.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import {useCart,useNavigate,useWishlist} from '../../customHooks'
+import {addItemToCart} from '../../dryProviders'
+const WishlistCard=({productObj})=>{
+  const navigate=useNavigate()
+  const [inCart,setButtonText]=useState(false)
+  const {dispatchCart,cartCount}=useCart()
+  const moveToCart=async(product)=>{
+    try{
+      
+      // const response=await axios.post('/api/user/cart',{product},{
```
Comment (tarunsankhla) [inline src/components/cards/wishlistCard.js:13]: Remove commented code

File: src/components/cart/index.css [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+.flex-H-center-VH{
+    display: flex;
+    align-items: center;
+    justify-content: center;
+}
+.quantity{
+    margin-left:0.5em;
+    margin-right:0.5em
```
Comment (tarunsankhla) [inline src/components/cart/index.css]: You could do is margin: 0 0.5em

File: src/components/cart/index.css [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    right: 0%;
+}
+
+/* break points */
+
+@media screen and (min-width:670px) {
+
+    .cart{
+        margin-left: 20%;
+        margin-right: 20%;
```
Comment (tarunsankhla) [inline src/components/cart/index.css]: Here too margin 0 20%

File: src/components/input/index.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import './index.css'
+// import {useAuth} from '../../contexts/login-context'
+
+const InputElement=(props)=>{
+    // const {dispatchAuth}=useAuth()
+
+    return <div>
+        {/* <label className="label">Email Address</label>
+        <input type="text" placeholder="jasonDeo.gmail.com" className="input" onChange={(e)=>dispatchAuth({type:"email",payload:e.target.value})}/>
```
Comment (tarunsankhla) [inline src/components/input/index.js]: Remove commented code

File: src/dryProviders/index.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+const removeItemFromWishlist=async(product,dispatchWishlist,setToWishlist,wishlistCount)=>{
+  try{
+    const response=await axios.delete(`/api/user/wishlist/${product._id}`,{headers:{authorization:"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI3MTUxN2NmOS05MTg4LTRlNGYtOWM1MS0xMzMxZWE1ZThkZmQiLCJlbWFpbCI6ImFkYXJzaGJhbGlrYUBnbWFpbC5jb20ifQ.WTkYnS_dAUXq8sBn-GKoX0BC6ZJKNpL8Q_CNUzlebJI"}})
+        dispatchWishlist({type:"wishlist",payload:{wishlist:response.data.wishlist,wishlistCount:wishlistCount-1,wishListed:false}})
+        setToWishlist(false)
+
+  }catch(e){
+    throw e
```
Comment (tarunsankhla) [inline src/dryProviders/index.js]: Handle error

File: src/routes/cartRoute.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+import { useCart } from '../contexts/cartContext'
+const Cart=()=>{
+    const {cart,checkoutPrice}=useCart()
+     
+      return <>
+       <Navbar/> 
+        <Headers/>
+        <hr/>
+         {cart.length!==0?cart.map((item)=> <CartProduct product={item}/>):<div className="text-align-center"><h1 >cart is empty</h1></div>}
+        <Checkout checkoutPrice={checkoutPrice}/>
```
Comment (tarunsankhla) [inline src/routes/cartRoute.js]: Format code for better readability

File: src/routes/productsRoute.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+      } )
+    },[])
+  
+      return <div className="products-section"> 
+      <Navbar/>
+      <div className="products">
+      <div>
+      <h2>Sort By Price</h2>     
+      <SortBy />
+      <h2>CATEGORIES</h2>
```
Comment (tarunsankhla) [inline src/routes/productsRoute.js]: Intendation format your code add vscode prettiesr extension

**Review by tarunsankhla — APPROVED** [review]
Overall it is great just minor changes of allignment and css so approving the pr

File: src/dryProviders/index.js [hunk 12]
```diff
@@ -0,0 +1,61 @@
+import axios from 'axios'
+
+const addItemToCart=async(product,dispatchCart,setButtonText,cartCount)=>{
+    try{
+      console.log(product,"entered intotry")
+      const response=await axios.post('/api/user/cart',{product},{
+        headers:{
+          authorization:"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI3MTUxN2NmOS05MTg4LTRlNGYtOWM1MS0xMzMxZWE1ZThkZmQiLCJlbWFpbCI6ImFkYXJzaGJhbGlrYUBnbWFpbC5jb20ifQ.WTkYnS_dAUXq8sBn-GKoX0BC6ZJKNpL8Q_CNUzlebJI"
```
Comment (pankaj-prog) [inline src/dryProviders/index.js]: store this authorization token in a variable.

File: src/contexts/wishlistContext.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+              return {...wishlistState}
+        }
+    }
+           
+    const [{wishlist,wishlistCount,},dispatchWishlist]=useReducer(reducerFn,{wishlist:[],wishlistCount:0})
+    
+    return <wishlistContext.Provider value={{wishlist,dispatchWishlist,wishlistCount,}}>
+    {children}
+    </wishlistContext.Provider>
+}
```
Comment (pankaj-prog) [inline src/contexts/wishlistContext.js]: give spaces in between to make code more readable, You can use prettier or some other tool.

File: src/routes/wishlistRoute.js [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+
+import axios from 'axios'
+import {useCart,useWishlist} from '../customHooks'
+
+const Wishlist=()=>{
+    const {wishlist}=useWishlist()
+   
+
+    return <>
+    <Navbar/>
```
Comment (pankaj-prog) [inline src/routes/wishlistRoute.js]: you can use navbar outside the routes, no need to render it again on changing of routes

File: src/routes/landingRoute.js [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+         {/* <TextOverlayCard/> */}
+         <h1 className="text-align-center">What we Deliver</h1>
+         <Carousel breakPoints={serviceCardBreakPoints} showArrows={false}> 
+         <ServiceCard/>
+         <ServiceCard/> 
+         <ServiceCard/> 
+        </Carousel> 
+        <h1 className="text-align-center">New Products</h1>    
+          <Carousel breakPoints={serviceCardBreakPoints} showArrows={false}>
+        {/* {featuredList.map((item)=><ProductCard productObj={item} />)} */}
```
Comment (pankaj-prog) [inline src/routes/landingRoute.js]: remove commented code

File: src/components/navbar/index.js [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+import { CgClose,CgMenu} from "react-icons/cg";
+import { VscHeart } from "react-icons/vsc";
+import {GrCart} from "react-icons/gr"
+import { useCart, useWishlist,useNavigate } from '../../customHooks';
+const Navbar = () => {
+const {sidebarToggle,setToggleState}=useSidebar()
+const {cartCount}=useCart()
+const {wishlistCount}=useWishlist()
+const navigate=useNavigate("/wishlist")
+
```
Comment (pankaj-prog) [inline src/components/navbar/index.js]: please indent your code

**Review by pankaj-prog — APPROVED** [review]

File: src/dryProviders/index.js [hunk 17]
```diff
@@ -0,0 +1,152 @@
...
+  dispatchWishlist,
+  setToWishlist,
+  wishlistCount
+) => {
+  try {
+    const response = await axios.delete(`/api/user/wishlist/${product._id}`, {
+      headers: {
+        authorization:
+          "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI3MTUxN2NmOS05MTg4LTRlNGYtOWM1MS0xMzMxZWE1ZThkZmQiLCJlbWFpbCI6ImFkYXJzaGJhbGlrYUBnbWFpbC5jb20ifQ.WTkYnS_dAUXq8sBn-GKoX0BC6ZJKNpL8Q_CNUzlebJI",
+      },
```
Comment (SinchanaMS) [inline src/dryProviders/index.js:97]: Store the token in a variable and pass it in the authorization header.

**Review by SinchanaMS — APPROVED** [review]

**Pull request merged**